src/main/java/com/monumental/services/UserService.java

```diff
@@ -84,46 +128,58 @@ public VerificationToken generateVerificationToken(User user, VerificationToken.
+    }
+
+    private void sendEmail(String recipientAddress, String name) {
+        this.sendEmail(recipientAddress, name, "");
+    }
+    
+    private void sendEmail(String recipientAddress, String name, String extraMessage) {
         SimpleMailMessage email = new SimpleMailMessage();
         email.setTo(recipientAddress);
         email.setFrom(outboundEmailAddress);
-        email.setSubject(subject);
-        email.setText(message + "\n\n" + confirmationUrl);
+        email.setSubject(this.messages.getMessage(name + ".subject", null, Locale.getDefault()));
+        email.setText(this.messages.getMessage(name + ".body", null, Locale.getDefault()) + extraMessage);
         mailSender.send(email);
     }
 
-    public void sendPasswordResetEmail(User user, VerificationToken token, String appUrl, Locale locale) {
-        String recipientAddress = user.getEmail();
-        String subject = "Reset your Monuments and Memorials password";
-        String confirmationUrl
-                = appUrl + "/password-reset/confirm?token=" + token.getToken();
-        String message = this.messages.getMessage("password-reset.begin", null, locale);
+    public void sendEmailChangeVerificationEmail(User user, VerificationToken token) {
+        this.sendEmail(
+            user.getEmail(),
+            "email-change.begin",
+            "\n\n" + this.publicUrl + "/account/update/confirm?token=" + token.getToken()
+        );
+    }
 
-        SimpleMailMessage email = new SimpleMailMessage();
-        email.setTo(recipientAddress);
-        email.setFrom(outboundEmailAddress);
-        email.setSubject(subject);
-        email.setText(message + "\n\n" + confirmationUrl);
-        mailSender.send(email);
+    public void sendSignupVerificationEmail(User user, VerificationToken token) {
+        this.sendEmail(
+            user.getEmail(),
+            "registration.success",
+            "\n\n" + this.publicUrl + "/signup/confirm?token=" + token.getToken()
+        );
     }
 
-    public void sendPasswordResetCompleteEmail(User user, String appUrl, Locale locale) {
-        String recipientAddress = user.getEmail();
-        String subject = "Your Monuments and Memorials password has been reset";
-        String message = this.messages.getMessage("password-reset.success", null, locale);
+    public void sendPasswordResetEmail(User user, VerificationToken token) {
+        this.sendEmail(
+            user.getEmail(),
+            "password-reset.begin",
+            "\n\n" +     this.publicUrl + "/password-reset/confirm?token=" + token.getToken()
+        );
+    }
 
-        SimpleMailMessage email = new SimpleMailMessage();
-        email.setTo(recipientAddress);
-        email.setFrom(outboundEmailAddress);
-        email.setSubject(subject);
-        email.setText(message);
-        mailSender.send(email);
+    public void sendPasswordResetCompleteEmail(User user) {
+        this.sendEmail(user.getEmail(), "password-reset.success");
     }
 }
```

Comment:
We should probably add some method docstrings to the methods in this class.

Reply:
Added